src/training/niah_probe.py

```
@@ -102,128 +110,151 @@ def should_probe(self, step: int) -> bool:
+                block.disable_memory = True
+
+            logits_nomem = model(input_ids)
+            logits_nomem = logits_nomem[:, :-1, :].contiguous()
+            loss_nomem = F.cross_entropy(
+                logits_nomem.view(-1, vocab_size), labels.view(-1)
+            )
+            ppl_nomem = torch.exp(loss_nomem).item()
 
-        # Determine pass/fail
+            # Restore original per-block memory flags
+            for block, flag in zip(model.blocks, orig_flags):
```

Ruff (0.14.14)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)